cryptus-app/components/wallet_viewer/nft_details/nft_details.tsx

```diff
@@ -20,14 +21,48 @@ type nft_details_props = {
 
 export default function NFTDetails(props: nft_details_props) {
   const { nft, rank } = props;
+  let properties = nft.properties;
+
+  let [collection_size, setCollection_size] = useState(Number);
+  let [rarity_rank, setraRity_rank] = useState(Number);
+
+  useEffect(() => {
+    const collectionCall = async () => {
+      try {
+        const collectionSize: number = await GetCollectionTokens(
```

Comment:
Etherscan API call gets blocked by adblocker (because the call includes “...contractADress...”

Etherscan API doesn’t return amount of nfts for erc-1155 tokens

Reply:
I know and I am working on an alternative on my rarity_rank task